diff --git a/src/platform/BUILD.gn b/src/platform/BUILD.gn
index 654875cde5..1472cf2623 100644
--- a/src/platform/BUILD.gn
+++ b/src/platform/BUILD.gn
@@ -311,6 +311,12 @@ if (chip_device_platform != "none" && chip_device_platform != "external") {
     } else if (chip_device_platform == "stm32") {
       device_layer_target_define = "STM32"
       defines += [ "CHIP_DEVICE_LAYER_TARGET=stm32" ]
+    } else if (chip_device_platform == "fermion") {
+      device_layer_target_define = "FERMION"
+      defines += [ 
+        "CHIP_DEVICE_LAYER_TARGET = fermion",
+        "CHIP_DEVICE_CONFIG_ENABLE_WIFI=${chip_enable_wifi}",
+      ]
     } else {
       device_layer_target_define = ""
     }
@@ -350,6 +356,7 @@ if (chip_device_platform != "none" && chip_device_platform != "external") {
       "OPEN_IOT_SDK",
       "ASR",
       "STM32",
+      "FERMION",
     ]
     foreach(possible_device_layer_target_define,
             possible_device_layer_target_defines) {
@@ -583,6 +590,8 @@ if (chip_device_platform != "none") {
       _platform_target = "ASR"
     } else if (chip_device_platform == "stm32") {
       _platform_target = "stm32"
+    } else if (chip_device_platform == "fermion") {
+      _platform_target = "fermion"
     } else {
       assert(false, "Unknown chip_device_platform: ${chip_device_platform}")
     }
