diff --git a/src/system/BUILD.gn b/src/system/BUILD.gn
index afeab8d343..d2c330715d 100644
--- a/src/system/BUILD.gn
+++ b/src/system/BUILD.gn
@@ -72,6 +72,8 @@ if (chip_device_platform == "cc13x2_26x2") {
   import("//build_overrides/bouffalolab_iot_sdk.gni")
 } else if (chip_device_platform == "bl702l") {
   import("//build_overrides/bouffalolab_iot_sdk.gni")
+} else if (chip_device_platform == "fermion") {
+  import("//build_overrides/fermion.gni")
 }
 
 buildconfig_header("system_buildconfig") {
@@ -205,6 +207,9 @@ source_set("system_config_header") {
   if (chip_device_platform == "mt793x") {
     public_deps += [ "${mt793x_sdk_build_root}:mt793x_sdk" ]
   }
+  if (chip_device_platform == "fermion") {
+    public_configs += [ "${fermion_platform_root}:fermion_config" ]
+  }
 }
 
 static_library("system") {
